src/jarabe/journal/journalactivity.py

```diff
@@ -488,5 +666,59 @@ def get_journal():
     return _journal
 
 
+def initialize_journal_object(title=None, bundle_id=None, 
+                                activity_id=None, project_metadata=None,
+                                icon_color=None,invited=False):
+
+    if not icon_color:
+        settings = Gio.Settings('org.sugarlabs.user')
+        icon_color = settings.get_string('color')
+
+    if not activity_id:
+        activity_id = activityfactory.create_activity_id()
+
+    if bundle_id == PROJECT_BUNDLE_ID:
+        logging.debug('[GSoC]initializing journal object for project %r' %[title, invited])
+        
+        jobject = datastore.create()
+        jobject.metadata['title'] = title
+        jobject.metadata['title_set_by_user'] = '0'
+        jobject.metadata['activity'] = PROJECT_BUNDLE_ID
+        jobject.metadata['activity_id'] = activity_id
```

Comment:
Can you share the shared lines between the PROJECT and non project if statements?